### ISSUE TITLE
Show hide blocks based functionality feature request

### ISSUE THREAD

**Issue by johnwbaxter** [weareferal/craft-matrix-field-preview#125]
Hi!

We have a use case where we need to show/hide matrix blocks depending on what user group they are.

You can do this in Matrix Mate which we are using at the moment, but we want to use Matrix Preview and would need similar functionality.

Is this possible via hooks if not something you'd want to implement?

Thanks!

**Comment by johnwbaxter** [comment]
Just to add to this, it would in general be great if the config for this plugin was offloaded to a config file and not stored in the DB!

**Comment by timmyomahony** [comment]
Hi @johnwbaxter thanks for raising. It sounds somewhat related to https://github.com/timmyomahony/craft-matrix-field-preview/issues/101. I’ll look into how MatrixMate handles this and certainly consider it.

Regarding the “config via file”, that’s definitely something I’m considering. Originally the idea was that different environments would want different previews, so the config was saved in the DB, but people seems to prefer being able to configure via the `project.yaml` so I’ll be moving over to that in a future version.

**Comment by johnwbaxter** [comment]
@timmyomahony Hey man, that's super cool, thank you! It's a really great plugin we've only just started using and as mentioned, we love Matrix Mate for the ability to apply logic to the config and show/hide stuff dependent on some business logic, so if this could do the same, then that'd be magic :)

**Comment by johnwbaxter** [comment]
@timmyomahony Cheeky request, but If you let out a minor release where you applied some classes to the parent html element of the block selector modal that included the handle of the current section it's on and optionally what member groups the person viewing is in, then we could use CP CSS in the interim to show hide blocks to achieve the same sorta thing in a vaguely hacky way :joy:

**Comment by timmyomahony** [comment]
Hi @johnwbaxter just to follow up on this, is this the functionality from MatrixMate that you're using to adjust blocks base on users:

https://github.com/vaersaagod/matrixmate?tab=readme-ov-file#advanced-configuration

**Comment by johnwbaxter** [comment]
@timmyomahony Yeah, that! As you can see, it lets us apply any logic which is super useful!
